From 5d6e7ddf559d9f2b80ae4a306e983a02dbc7f67d Mon Sep 17 00:00:00 2001
From: Murat Sezgin <msezgin@codeaurora.org>
Date: Tue, 30 Mar 2021 18:59:51 -0700
Subject: [PATCH] net: Add DSCP remark set flags

These flags indicates that which QoS features are set
in the DSCP conntrack extension. DSCP, Priority, or IGS.

Signed-off-by: Murat Sezgin <msezgin@codeaurora.org>
Change-Id: I11b03c683f57e6208550079f187fa94d6f7b616b
---
 include/net/netfilter/nf_conntrack_dscpremark_ext.h | 7 +++++++
 1 file changed, 7 insertions(+)

--- a/include/net/netfilter/nf_conntrack_dscpremark_ext.h
+++ b/include/net/netfilter/nf_conntrack_dscpremark_ext.h
@@ -28,6 +28,13 @@
 #define NF_CT_DSCPREMARK_EXT_RULE_VALID 0x1
 #define NF_CT_DSCPREMARK_EXT_RULE_NOT_VALID 0x0
 
+/* Which QoS features are set flags */
+#define NF_CT_DSCPREMARK_EXT_PRIO 0x1
+#define NF_CT_DSCPREMARK_EXT_DSCP 0x2
+#define NF_CT_DSCPREMARK_EXT_IGS_QOS 0x4
+#define NF_CT_DSCPREMARK_EXT_MARK 0x8
+#define NF_CT_DSCPREMARK_EXT_SAWF 0x10
+
 /*
  * DSCP remark conntrack extension structure.
  */
@@ -39,6 +46,8 @@ struct nf_ct_dscpremark_ext {
 	__u8 flow_dscp;		/* IP DSCP value for original direction */
 	__u8 reply_dscp;	/* IP DSCP value for reply direction */
 	__u16 rule_flags;	/* Rule Validity flags */
+	__u16 flow_set_flags;	/* Original direction set flags */
+	__u16 return_set_flags;	/* Reply direction set flags */
 };
 
 /*
--- a/include/uapi/linux/in.h
+++ b/include/uapi/linux/in.h
@@ -62,6 +62,8 @@ enum {
 #define IPPROTO_MTP		IPPROTO_MTP
   IPPROTO_BEETPH = 94,		/* IP option pseudo header for BEET	*/
 #define IPPROTO_BEETPH		IPPROTO_BEETPH
+  IPPROTO_ETHERIP = 97,
+#define IPPROTO_ETHERIP		IPPROTO_ETHERIP 
   IPPROTO_ENCAP = 98,		/* Encapsulation Header			*/
 #define IPPROTO_ENCAP		IPPROTO_ENCAP
   IPPROTO_PIM = 103,		/* Protocol Independent Multicast	*/
